tests/e2e/api_event.go

```diff
@@ -0,0 +1,46 @@
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/sassoftware/event-provenance-registry/pkg/storage"
+)
+
+const eventURI = "http://localhost:8042/api/v1/events/"
+
+type postEventResponse struct {
+	// ID of the created event
+	Data   string
+	Errors []string
+}
+
+type getEventResponse struct {
+	Data   []storage.Event
+	Errors []string
+}
+
+type eventInput struct {
```

Comment:
I feel like we should use the same data structures as in the library. I realize that because of the graphql data type junk you did this. Eventually, I think we should use the same things though.

Reply:
The json type definitely wasn't clean to work with. I wrote the `toPayload()` funcs intentionally so that we could test invalid input without needing.

I prefer, at least for now, to use these standalone structs so its clear what our contracts are. I have several concerns with the lib types that are out of the scope of this PR:
- `pkg/api/rest/Response` has type `any` for the data field. This is fine but I'd rather not handle type casting in tests
- Lib lacks separate input structs, it just decodes input into the desired type & sends it to the db layer. Absolutely a bug but for example, you can post an event with a custom `created_at` field and it will use that instead of the server populating that field

If you feel strongly that we should use the same structs then I can revisit this now, otherwise I'd like that to be a follow up task after #70 .

Reply:
It was more of a "lets keep it in mind for later."